modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/AbstractWorkflowOperationHandler.java

```diff
@@ -360,6 +378,160 @@ protected String getConfig(WorkflowOperationInstance woi, String key) throws Wor
+              throw new WorkflowOperationException(elem + " is not a valid flavor!");
+            }
+          }
+        }
+        break;
+      default:
+        throw new WorkflowOperationException("Couldn't process srcFlavors configuration option!");
+    }
+    tagsAndFlavors.setSrcFlavors(srcFlavorList);
+
+    List<String> targetTagList = new ArrayList<>();
+    String targetTag;
+    switch(targetTags) {
+      case none:
+        break;
+      case one:
+        targetTag = StringUtils.trimToNull(woi.getConfiguration(TARGET_TAG));
+        if (targetTag == null) {
+          throw new WorkflowOperationException("Configuration key '" + TARGET_TAG + "' must be set");
+        } else {
+          targetTagList.add(targetTag);
+        }
+        break;
+      case many:
+        targetTagList = asList(StringUtils.trimToNull(woi.getConfiguration(TARGET_TAGS)));
+        targetTag = StringUtils.trimToNull(woi.getConfiguration(TARGET_TAG));
+        if (targetTagList.isEmpty() && targetTag != null) {
+          targetTagList.add(targetTag);
+        }
+        if (targetTagList.isEmpty()) {
+          logger.warn("Configuration keys '" + TARGET_TAGS + "and" + TARGET_TAG + "' are empty");
+        }
+        break;
+      default:
+        throw new WorkflowOperationException("Couldn't process target-tag configuration option!");
+    }
+    tagsAndFlavors.setTargetTags(targetTagList);
+
+    List<String> targetFlavorList = new ArrayList<>();
+    String targetFlavor;
+    switch(targetFlavors) {
+      case none:
+        break;
+      case one:
+        targetFlavor = StringUtils.trimToNull(woi.getConfiguration(TARGET_FLAVOR));
+        if (targetFlavor == null) {
+          throw new WorkflowOperationException("Configuration key '" + TARGET_FLAVOR + "' must be set");
+        } else {
+          targetFlavorList.add(targetFlavor);
+          try {
+            flavor = MediaPackageElementFlavor.parseFlavor(targetFlavorList.get(0));
+          } catch (IllegalArgumentException e) {
+            throw new WorkflowOperationException(targetFlavorList.get(0) + " is not a valid flavor!");
+          }
+        }
+        break;
+      case many:
+        targetFlavorList = asList(StringUtils.trimToNull(woi.getConfiguration(TARGET_FLAVORS)));
+        targetFlavor = StringUtils.trimToNull(woi.getConfiguration(TARGET_FLAVOR));
+        if (targetFlavorList.isEmpty() && targetFlavor != null) {
+          targetFlavorList.add(targetFlavor);
+        }
+        if (targetTagList.isEmpty()) {
+          logger.warn("Configuration key '" + TARGET_FLAVORS + "and" + TARGET_FLAVOR + "' are empty");
+        } else {
+          for (String elem : targetFlavorList) {
+            try {
+              flavor = MediaPackageElementFlavor.parseFlavor(elem);
+            } catch (IllegalArgumentException e) {
+              throw new WorkflowOperationException(elem + " is not a valid flavor!");
+            }
+          }
+        }
```

Comment:
We don't need to check plus you check `targetTagList`:

```suggestion
        for (String elem : targetFlavorList) {
          try {
            flavor = MediaPackageElementFlavor.parseFlavor(elem);
          } catch (IllegalArgumentException e) {
            throw new WorkflowOperationException(elem + " is not a valid flavor!");
          }
        }
```